saleor/order/views.py

```diff
@@ -141,12 +141,12 @@ def create_password(request, token):
     else:
         login_form = None
     if register_form.is_valid():
-        register_form.save(request)
-        password = form_data.get('password1')
-        auth_user = auth.authenticate(email=email, password=password)
-        if auth_user is not None:
-            auth.login(request, auth_user)
-        attach_order_to_user(order, auth_user)
+        register_form.save()
+        password = form_data.get('password')
+        user = auth.authenticate(email=email, password=password)
+        if user:
```

Comment:
You're right - user won't be authenticated only when there is an issue with form. I'll remove `if user:` in a minute.